### PR TITLE
List view styling

### PR DESCRIPTION
**Pull request by ciradu2204**
## Description

This PR removes the background color on the whole list item and adds it on the checkbox only. It also categorizes the items in different groups: soon, kind of soon, inactive, and not soon. 

## Related Issue
#31 

## Acceptance Criteria
- [x] Add a background color to the checkbox only
- [x] Group the items into categories based on how soon they are likely to be bought
- [x] Add a title to each category
- [x] Use appropriate styling for the titles and item names

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|   ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Testing Steps / QA Criteria

1. From your terminal pull down this branch with `git pull list-view-styling` and check that branch out with `git checkout list-view-styling.`
2. Then `npm start` to launch the app.
3. Create a `new shopping list `or` join with an existing token.`
4. Add items using the add view. Ensure that the shopping list has at least 4 items. 
6. Find a document in the [database](https://console.firebase.google.com/u/1/project/tcl-25-shopping-list/firestore/data~2Flists) with your token.
7.  Make the first item `DaysLeftForTheNextPurchaseDate` 's value to be bellow 7 and `numberOfPurchases`'s to be above 2.
8. Make the second item `DaysLeftForTheNextPurchaseDate` 's value to be between 7 and 30 and `numberOfPurchases`'s to be above 2.
9. Make the third item `DaysLeftForTheNextPurchaseDate` 's value to be above 30 and `numberOfPurchases`'s to be above  2.
10. Make the fourth item `numberOfPurchases`  's value to be bellow 2.
11. Return to the app's home screen. Check out the list view to see how the four items are organized into groups, each with a different checkbox background colour.

### CODE REVIEW

**Review by katia-utochkina — APPROVED** [review]

File: src/lib/capitalize.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+const capitalizeFirstLetter = (item) =>
+  item.replace(/(^\w{1})|(\s+\w{1})/g, (letter) => letter.toUpperCase());
+
+export default capitalizeFirstLetter;
```
Comment (alexlsalt) [inline src/lib/capitalize.js:4]: Love the idea of abstracting this out rather than cluttering the main code with it 👍

**Review by alexlsalt — APPROVED** [review]
Good work on this feature! The testing steps were super clear and it was a joy to see the final effect of it! 🎉 

I think everything looks good on my end 👍 I'm interesting to see Golfo's review as well.

Just out of curiosity (and sorry if this is already explained somewhere else), but why remove the ariaLabel?

**Comment by ciradu2204** [comment]
> Good work on this feature! The testing steps were super clear and it was a joy to see the final effect of it! 🎉
> 
> I think everything looks good on my end 👍 I'm interesting to see Golfo's review as well.
> 
> Just out of curiosity (and sorry if this is already explained somewhere else), but why remove the ariaLabel?

Thanks, Alex❤️. I just realised it's a mistake I made. At first, I thought it was the one causing some css errors; that's why I removed it. But I figured that it was not the one.

**Comment by faysvas** [comment]
I like how the list is split up in smaller parts, it really helps you understand when you need to buy something
image
After adding items I get this view. I click on newly added items but nothing happens. I'm not really sure if I bought an item or not :sweat_smile:

**Comment by faysvas** [comment]
Really nice touch adding a link to the database in the testing steps :100: !

**Comment by ciradu2204** [comment]
> I like how the list is split up in smaller parts, it really helps you understand when you need to buy something
> image
> After adding items I get this view. I click on newly added items but nothing happens. I'm not really sure if I bought an item or not 😅

Thanks Golfo, based on the ac description -> Inactive (when there’s only 1 purchase in the database or the purchase is really out of date [the time that has elapsed since the last purchase is 2x what was estimated]). So, when it's your first purchase, it won't move to another category. But I can change to be 'and' not 'or'

**Review by YuriDevAT — APPROVED** [review]
Great idea to separate the items into different categories, this makes it way easier to understand than only colors 👍🏽 Nice work!! 😁 

**Suggestions:**
- svg/img could be stored in a folder
- when index.css is not used, it could be deleted (to prevent misunderstandings)

**Attention:**
- when an item is too long (without any whitespaces) a scrollbar appears and the list is quite not readable in mobile view anymore
- when the list itself is too long, the nav links are not the last part of the list anymore but somewhere between


Screenshot 2021-05-22 at 20 48 30

Screenshot 2021-05-22 at 20 13 43

**Pull request merged**